bindings/java/src/org/sleuthkit/datamodel/Blackboard.java

```diff
@@ -0,0 +1,176 @@
+	 * Gets both static and dynamic IDs.
+	 *
+	 * @param ds_obj_id  data source obj id 
+	 * 
+	 * @return The list of unique IDs
+	 *
+	 * @throws TskCoreException exception thrown if a critical error occurred
+	 *                          within tsk core
+	 */
+	public List<BlackboardArtifact.Type> getArtifactTypesInUseByDataSource(long ds_obj_id) throws TskCoreException {
+		
+		final String queryString = "SELECT DISTINCT arts.artifact_type_id AS artifact_type_id, "
+					+ "types.type_name AS type_name, types.display_name AS display_name "
+					+ "FROM blackboard_artifact_types AS types "
+					+ "INNER JOIN blackboard_artifacts AS arts "
+					+ "ON arts.artifact_type_id = types.artifact_type_id "
+					+ "WHERE arts.data_source_obj_id = " + ds_obj_id;
+				
+		SleuthkitCase.CaseDbConnection connection = caseDb.getConnection();
+		caseDb.acquireSingleUserCaseReadLock();
+		Statement statement = null;
+		ResultSet resultSet = null;
+		try {
+			statement = connection.createStatement();
+			resultSet = connection.executeQuery(statement, queryString); //NON-NLS
+
+			List<BlackboardArtifact.Type> uniqueArtifactTypes = new ArrayList<BlackboardArtifact.Type>();
+			while (resultSet.next()) {
+				uniqueArtifactTypes.add(new BlackboardArtifact.Type(resultSet.getInt("artifact_type_id"),
+						resultSet.getString("type_name"), resultSet.getString("display_name")));
+			}
+			return uniqueArtifactTypes;
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error getting artifact types is use for data source." + ex.getMessage(), ex);
+		} finally {
+			SleuthkitCase.closeResultSet(resultSet);
+			SleuthkitCase.closeStatement(statement);
+			connection.close();
+			caseDb.releaseSingleUserCaseReadLock();
+		}
+		
+	}
+	
+	/**
+	 * Get count of all blackboard artifacts of a given type for the given
+	 * Data source. Does not include rejected artifacts.
+	 *
+	 * @param artifactTypeID artifact type id (must exist in database)
+	 * @param ds_obj_id         data source object id
+	 *
+	 * @return count of blackboard artifacts
+	 *
+	 * @throws TskCoreException exception thrown if a critical error occurs
+	 *                          within TSK core
+	 */
+	public long getBlackboardArtifactsCountByDataSource(int artifactTypeID, long ds_obj_id) throws TskCoreException {
+		return getArtifactsCountHelper(artifactTypeID, 
+				"blackboard_artifacts.data_source_obj_id = '" + ds_obj_id + "';");
+	}
+	
+	/**
+	 * Get all blackboard artifacts of a given type. Does not included rejected
+	 * artifacts.
+	 *
+	 * @param artifactTypeID artifact type to get 
+	 * @param ds_obj_id data source to look under
+	 * 
+	 * @return list of blackboard artifacts
+	 *
+	 * @throws TskCoreException exception thrown if a critical error occurs
+	 *                          within TSK core
+	 */
+	public List<BlackboardArtifact> getBlackboardArtifactsByDataSource(int artifactTypeID, long ds_obj_id) throws TskCoreException {
```

Comment:
This method name can be shortened by removing "Blackboard" since we are in the Blackboard class. We could probably also shorten the method name by dropping the "ByDataSource" part, allowing the dataSourceObjId to distinguish overloads of a getArtifacts when we move SleuthkitCase.getBlackboardArtifacts into this class.